doc/tools/docgram/doc_grammar.ml

```diff
@@ -1754,7 +1757,7 @@ let process_rst g file args seen tac_prods cmd_prods =
           mtch (* update cmd/tacn *)
     in
     let map = ref seen_map in
-    if NTMap.mem first_rhs !map then
+    if NTMap.mem first_rhs !map && !show_warn then
```

Comment:
Having the `show_warn` test first would avoid computing the other condition when it's not used.

Reply:
That was a stylistic choice--I wanted to have the interesting condition first and the boilerplate `!show_warn` afterwards.  The performance impact is infintesmal.